include/yarp/parser.h

```diff
@@ -390,6 +376,32 @@ struct yp_parser {
     // communicate this information. So we store it here and pass it through
     // when we find tokens that we need it for.
     yp_node_flags_t integer_base;
+
+    // Whether or not we're at the beginning of a command
+    unsigned int command_start : 1;
+
+    // Whether or not we're currently recovering from a syntax error
+    unsigned int recovering : 1;
+
+    // Whether or not the encoding has been changed by a magic comment. We use
+    // this to provide a fast path for the lexer instead of going through the
+    // function pointer.
+    unsigned int encoding_changed : 1;
```

Comment:
Because this is performance-sensitive it seems worth to have its own 8-bits/byte.
Otherwise it means having to use an extra `& mask` whenever checking this.

Reply:
TIL, that makes a lot of sense. Updated.